discovery-handlers/opcua/src/discovery_impl.rs

```diff
@@ -193,29 +193,39 @@ fn get_socket_addr(url: &str) -> Result<SocketAddr, anyhow::Error> {
 }
 
 // This checks if the discovery_url can be resolved, if not use ip address instead
-fn get_discovery_url_ip(ip_url: &str, discovery_url: String) -> Result<String, anyhow::Error> {
-    let url = Url::parse(&discovery_url)
-        .map_err(|_| anyhow::format_err!("could not parse url {discovery_url}"))?;
-    if url.scheme() != OPC_TCP_SCHEME {
+fn get_discovery_url_ip(
+    ip_url_str: &str,
+    discovery_url_str: String,
+) -> Result<String, anyhow::Error> {
+    let ip_url = Url::parse(ip_url_str)
+        .map_err(|_| anyhow::format_err!("could not parse url {ip_url_str}"))?;
```

Comment:
Mapping errors can be simplified by importing `anyhow::Error` and doing the following
```suggestion
        .with_context(|_| "could not parse url {ip_url_str}")?;
```